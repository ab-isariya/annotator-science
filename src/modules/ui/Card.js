import PropTypes from 'prop-types';

const Card = ({children, className, padding, active, hover, ...otherProps}) => (
  <div
    className={`border border-grey-100 rounded-md  ${padding || 'p-4'} 
      ${active ? 'filter-active bg-grey-50' : 'bg-white'}
      ${hover ? 'hover:bg-grey-50' : ''}
      ${className}`}
    {...otherProps}>
    {children}
  </div>
);

Card.propTypes = {
  //Classnames from TailwindCSS
  className: PropTypes.string,
  //Children rendered inside of container
  children: PropTypes.node,
  //Extraneous props attached to the container for things like onClick, onHover, ect.
  otherProps: PropTypes.any,
  //Padding Classname from tailwind to use
  //NOTE(Rejon): Some cards require scrolling that can get too cramped using initial p-4
  //             padding. This will enable manually setting padding for children elements.
  padding: PropTypes.string,
  //If this card should have the active style or not.
  active: PropTypes.bool,
  //If this card should use hover styles
  hover: PropTypes.bool
};

export default Card;
