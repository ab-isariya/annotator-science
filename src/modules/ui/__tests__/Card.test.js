import Card from '../Card';
import renderer from 'react-test-renderer';

describe('<Card/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<Card />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with children', () => {
    const comp = renderer.create(<Card>Hello</Card>).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children).not.toEqual(null);
  });

  it('Renders with a custom classname', () => {
    const comp = renderer.create(<Card className="testClass" />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });

  it('Renders with a custom padding', () => {
    const comp = renderer.create(<Card padding="px-3" />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/px-3/i);
  });

  it('Renders with active classes', () => {
    const comp = renderer.create(<Card active />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/filter-active bg-grey-50/i);
  });

  it('Renders with hover classes', () => {
    const comp = renderer.create(<Card hover />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/hover:bg-grey-50/i);
  });
});
