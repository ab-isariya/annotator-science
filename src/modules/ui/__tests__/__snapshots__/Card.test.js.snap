// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Renders with a custom classname 1`] = `
<div
  className="border border-grey-100 rounded-md  p-4 
      bg-white
      
      testClass"
/>
`;

exports[`<Card/> Renders with a custom padding 1`] = `
<div
  className="border border-grey-100 rounded-md  px-3 
      bg-white
      
      undefined"
/>
`;

exports[`<Card/> Renders with active classes 1`] = `
<div
  className="border border-grey-100 rounded-md  p-4 
      filter-active bg-grey-50
      
      undefined"
/>
`;

exports[`<Card/> Renders with children 1`] = `
<div
  className="border border-grey-100 rounded-md  p-4 
      bg-white
      
      undefined"
>
  Hello
</div>
`;

exports[`<Card/> Renders with hover classes 1`] = `
<div
  className="border border-grey-100 rounded-md  p-4 
      bg-white
      hover:bg-grey-50
      undefined"
/>
`;

exports[`<Card/> Renders without crashing 1`] = `
<div
  className="border border-grey-100 rounded-md  p-4 
      bg-white
      
      undefined"
/>
`;
